src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2067,6 +2078,25 @@ public int getSSTableLevel()
         return sstableMetadata.sstableLevel;
     }
 
+    /**
+     * Mutate Stats Metadata with a lock to avoid racing with entire-sstable-streaming and then reload sstable metadata
+     *
+     * @param transform function to mutate stats metadata
+     */
+    public void mutateAndReloadStats(UnaryOperator<StatsMetadata> transform) throws IOException
+    {
+        tidy.global.writeLock.lock();
+        try
+        {
+            descriptor.getMetadataSerializer().mutate(descriptor, transform);
+            reloadSSTableMetadata();
+        }
+        finally
+        {
+            tidy.global.writeLock.unlock();
+        }
```

Comment:
@jasonstack Let's move this block to `runWithWriteLock()`. It might help future readers looking at usages of `runWithWriteLock()` trace back to the stats writing code itself, rather than having to search on usages of the write lock itself.

Reply:
+1, missed it when reverting code..